bugzoo/cli/controllers/tool.py

```diff
@@ -0,0 +1,98 @@
+        arguments=[
+            (['-q'], {'help': 'prints an unannotated list of the names of all registered tools',  # noqa: pycodestyle
+                      'action': 'store_true'}),
+        ]
+    )
+    def list(self) -> None:
+        headers = ['Tool', 'Source', 'Installed?']
+        data = []
+        tools = self.app.daemon.tools
+        for tool in tools:
+            row = [
+                tool.name,
+                tool.source if tool.source else '-',
+                'Yes' if tools.is_installed(tool) else 'No'
+            ]
+            data.append(row)
+        data = sorted(data, key=operator.itemgetter(1, 0))
+        if self.app.pargs.q:
+            print('\n'.join([r[0] for r in data]))
+        else:
+            self.app.render(data, headers=headers)
+
+    @cement.ex(
+        help='downloads the Docker image for a given tool',
+        arguments=[(['tool'], {'help': 'the name of the tool', 'type': str})]
+    )
+    def download(self) -> None:
+        name_tool = self.app.pargs.tool  # type: str
+        print('downloading tool: {}'.format(name_tool))
+        tools = self.app.daemon.tools
+        try:
+            tool = tools[name_tool]
+        except KeyError:
+            raise exceptions.ToolNotFound(name_tool)
+        tools.download(tool)
+
+    @cement.ex(
+        help='uploads the Docker image for a given tool',
+        arguments=[(['tool'], {'help': 'the name of the tool', 'type': str})]
+    )
+    def upload(self) -> None:
+        name_tool = self.app.pargs.tool  # type: str
+        print('uploading tool: {}'.format(name_tool))
+        tools = self.app.daemon.tools
+        tool = tools[name_tool]
+        tools.upload(tool)
+
+    @cement.ex(
+        help='builds the Docker image for a given bug',
+        arguments=[(['tool'], {'help': 'the name of the tool', 'type': str})]
+    )
+    def build(self) -> None:
+        name_tool = self.app.pargs.tool  # type: str
+        print('building tool: {}'.format(name_tool))
+        tools = self.app.daemon.tools
+        try:
+            tool = tools[name_tool]
+        except KeyError:
+            raise exceptions.ToolNotFound(name_tool)
+        tools.build(tool)
+
+    @cement.ex(
+        help='uninstalls the Docker image for a given bug',
+        arguments=[
+            (['tool'], {'help': 'the name of the tool', 'type': str}),
+            (['-f', '--force'],
+             {'help': 'does not report an error if image is not installed.',
+              'action': 'store_true'})
+        ]
+    )
+    def uninstall(self) -> None:
+        force = self.app.pargs.force  # type: bool
+        name_tool = self.app.pargs.tool  # type: str
+        print('uninstalling tool: {}'.format(name_tool))
+        tools = self.app.daemon.tools
+        try:
+            tool = tools[name_tool]
+        except KeyError:
+            raise exceptions.ToolNotFound(name_tool)
+        tools.uninstall(tool, force=force)
```

Comment:
Does this also removes the image? I was never a fan of `build` and `uninstall`. It should either be `install` and `uninstall` or `build` and `remove`.

Reply:
Fair point. So, we don't use `install` any more because it's not clear whether the image is being downloaded or built from scratch. That does make `uninstall` a weird verb; `remove` is a better alternative. :+1:

Reply:
Okay. I've renamed `tool uninstall` with `tool rm` and `tool remove`, and `bug uninstall` with `bug rm` and `bug remove`.